scripts/setup.sh

```diff
@@ -11,6 +11,7 @@ if [ -f "${my_venv}/bin/activate" ]; then
     # shellcheck disable=SC1091
     . "${my_venv}/bin/activate"
     # Install commit requirements
+    python3 -m ensurepip --upgrade
```

Comment:
you can better use --seed in the venv creation
https://docs.astral.sh/uv/reference/cli/#uv-venv

Reply:
Good call, goes for the Smile stuff too - we might need to re-visit all of them